### PR TITLE
Add support for specifying the RunAction `launchStyle`

### PR DESCRIPTION
**Pull request by baekteun**
Resolves https://github.com/tuist/tuist/issues/3511

### Short description 📝

Added a new property to RunAction called `launchStyle`

### How to test the changes locally 🧐

1. Generate a project with defined `launchStyle` for RunAction.
2. Go to `Edit scheme` -> `Run` and verify if the `Launch` On field is correctly set.

### Contributor checklist ✅

- [x] The code has been linted using run `make lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/ProjectDescription/RunAction.swift [hunk 1]
```diff
@@ -34,6 +34,9 @@ public struct RunAction: Equatable, Codable {
     /// A target that will be used to expand the variables defined inside Environment Variables definition (e.g. $SOURCE_ROOT)
     public let expandVariableFromTarget: TargetReference?
 
+    /// The launch style of the action
+    public let launchStyle: LaunchStyle?
```
Comment (danyf90) [inline Sources/ProjectDescription/RunAction.swift:38]: And set the default in the init
```suggestion
    public let launchStyle: LaunchStyle
```

File: Sources/TuistGraph/Models/RunAction.swift [hunk 2]
```diff
@@ -15,6 +15,7 @@ public struct RunAction: Equatable, Codable {
     public let options: RunActionOptions
     public let diagnosticsOptions: Set<SchemeDiagnosticsOption>
     public let expandVariableFromTarget: TargetReference?
+    public let launchStyle: LaunchStyle?
```
Comment (danyf90) [inline Sources/TuistGraph/Models/RunAction.swift:18]:
```suggestion
    public let launchStyle: LaunchStyle
```

**Review by danyf90 — COMMENTED** [review]
Aren't we missing the mapping in `RunAction+ManifestMapper.swift`?

**Comment by baekteun** [comment]
Thank you @danyf90 !

I pushed

- Changed the type of the launchStyle property of the RunAction structure from `LaunchStyle?` -> `LaunchStyle`.
- Changed default value of launchStyle from `nil` -> `.automatically` in functions that create RunAction or init

**Comment by danyf90** [comment]
@allcontributors add @baekteun for code

File: Sources/TuistGraph/Models/LaunchStyle.swift [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
     case waitForExecutableToBeLaunched
+
+    public static func from(launchStyle: ProjectDescription.LaunchStyle) -> TuistGraph.LaunchStyle {
+        switch launchStyle {
+        case .automatically:
+            return .automatically
+        case .waitForExecutableToBeLaunched:
+            return .waitForExecutableToBeLaunched
+        }
+    }
```
Comment (danyf90) [inline Sources/TuistGraph/Models/LaunchStyle.swift:15]: I would move this to `Sources/TuistLoader/Models+ManifestMappers/RunAction+ManifestMapper.swift` directly.
`TuistGraph` shouldn't depend on `ProjectDescription`

Comment (baekteun) [reply Sources/TuistGraph/Models/LaunchStyle.swift]: I tried placing the function that maps a ProjectDescription.LaunchStyle to TuistGraph.LaunchStyle in `TuistLoader/Models.ManifestMappers/LaunchStyle+ManifestMapper.swift`, is that okay?

**Review by danyf90 — APPROVED** [review]

**Pull request merged**